common/step_provision.go

```diff
@@ -46,6 +46,16 @@ func PopulateProvisionHookData(state multistep.StateBag) map[string]interface{}
 		// Warn user that the id isn't implemented
 		hookData["ID"] = "ERR_ID_NOT_IMPLEMENTED_BY_BUILDER"
 	}
+
+	// source_image_name is placed in state by builders
+	// and it's currently implemented only in Amazon
+	imageName, ok := state.GetOk("source_image_name")
+	if ok {
```

Comment:
nit: the else statement can be the default value. What do you think about dropping the else statement? 

```go
hookData["SourceImageName"] = "ERR_SOURCE_IMAGE_NAME_NOT_IMPLEMENTED_BY_BUILDER"                                                                                                          
if imageName, ok := state.GetOk("source_image_name"); ok {                                                                                                                              
  hookData["SourceImageName"] = imageName 
}